python/cuspatial/cuspatial/tests/test_point_linestring_distance.py

```diff
@@ -167,3 +167,121 @@ def test_mixed_geometry_series_raise(lhs, rhs):
 
     with pytest.raises(ValueError, match=".*must contain only.*"):
         pairwise_point_linestring_distance(lhs, rhs)
+
+
+def test_multipoint_multilinestring_sliced_pairs(
+    multipoint_generator, multilinestring_generator
+):
+    num_pairs = 2
+    max_num_linestring_per_multilinestring = 2
+    max_num_segments_per_linestring = 2
+    max_num_points_per_multipoint = 2
+
+    hpts = gpd.GeoSeries(
+        [*multipoint_generator(num_pairs, max_num_points_per_multipoint)]
+    )
+    hlines = gpd.GeoSeries(
+        [
+            *multilinestring_generator(
+                num_pairs,
+                max_num_linestring_per_multilinestring,
+                max_num_segments_per_linestring,
+            )
+        ]
+    )
+
+    gpts = from_geopandas(hpts)
+    glines = from_geopandas(hlines)
+
+    hslicepts = hpts[0:1]
+    hslicelines = hlines[0:1]
+    slicegpts = gpts[0:1]
+    sliceglines = glines[0:1]
+    got = pairwise_point_linestring_distance(slicegpts, sliceglines)
+    expected = hslicepts.distance(hslicelines)
+
+    cudf.testing.assert_series_equal(got, cudf.Series(expected))
+
+    hslicepts = hpts[1:2]
+    hslicelines = hlines[1:2]
+    slicegpts = gpts[1:2]
+    sliceglines = glines[1:2]
+    got = pairwise_point_linestring_distance(slicegpts, sliceglines)
+    expected = hslicepts.distance(hslicelines)
+
+    # Just dropping the index for now, that is not libcuSpatial's problem
+    got.index = [1]
+    cudf.testing.assert_series_equal(got, cudf.Series(expected))
+
+
+@pytest.mark.parametrize("slice_index", [0, 1, 2, 3, 4])
+def test_multipoint_multilinestring_sliced_many(
+    multipoint_generator, multilinestring_generator, slice_twenty, slice_index
+):
+    num_pairs = 20
+    max_num_linestring_per_multilinestring = 5
+    max_num_segments_per_linestring = 5
+    max_num_points_per_multipoint = 5
+
+    hpts = gpd.GeoSeries(
+        [*multipoint_generator(num_pairs, max_num_points_per_multipoint)]
+    )
+    hlines = gpd.GeoSeries(
+        [
+            *multilinestring_generator(
+                num_pairs,
+                max_num_linestring_per_multilinestring,
+                max_num_segments_per_linestring,
+            )
+        ]
+    )
+
+    gpts = from_geopandas(hpts)
+    glines = from_geopandas(hlines)
+
+    pslice = slice_twenty[slice_index]
+    hslicepts = hpts[pslice]
+    hslicelines = hlines[pslice]
+    slicegpts = gpts[pslice]
+    sliceglines = glines[pslice]
+    got = pairwise_point_linestring_distance(slicegpts, sliceglines)
+    expected = hslicepts.distance(hslicelines)
+    got.index = cudf.RangeIndex(pslice.start, pslice.stop)
```

Comment:
Looks like the python API is not copying the indices correctly.

Reply:
That's correct. I could modify the python API here, but I expect this to be an inner method on `object.distance(object)` eventually, so I didn't make those changes yet. Want me to?

Reply:
Nope, this is out of scope of this PR. But should track with an issue.